ui/apps/platform/src/Containers/Collections/RuleSelector/RuleSelector.tsx

```diff
@@ -49,12 +47,14 @@ function RuleSelector({
         }
 
         const selectorMap: Record<RuleSelectorOption, ScopedResourceSelector> = {
-            All: {},
+            All: { type: 'All' },
             ByName: {
+                type: 'ByName',
```

Comment:
Not asking you to change anything but was curious of your opinion on having these 3 types (All, ByName, ByLabel) defined as constants?

Reply:
This is something I go back and forth on, but lately I've been just writing the string directly in cases where the code is well-typed enough that a mistyped string would result in a compile time error. This might be worth bringing up during one of the weekly calls though to see if the team has a strong preference one way or another.

My initial feeling is the the pros/cons for using a constant here are:
**Pros** 
- changing the property name only requires a change in one place
- easier to Ctrl+Click to see where the relevant constants are coming from

**Cons**
- more verbose (subjective)
- changing the expected value _doesn't_ result in compile errors at each place the string is used. I could see an argument for this being a trail of breadcrumbs to get eyes on relevant code during a refactor.

Reply:
Ah - and didn't realize there were too many changes in the parser for git to detect it as a rename instead of an entirely new file... I can swap the name back if we think it is important to retain that history at this point.

Reply:
TypeScript string enumeration or discriminated union seem fluent as literal strings.

Reply:
If we felt the need to encapsulate, then Redux action creator **functions** is a precedent to adapt.